script.js

```diff
@@ -1,4 +1,193 @@
+//global variables
+let playerTurnCounter = 1;
+const playerDice = [];
+let totalPlayer = 0;
+let numOfDice = 0;
+let playerScore = [];
+let autoModeFlag = 1;
+let leaderBoard = [];
+let reverseModeFlag = 0;
+let knockOutFlag = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (input.toLowerCase().includes("reverse")) {
+    reverseModeFlag = 1;
+    return "Reversed mode activated. Please get the lowest score to win";
+  }
+  if (input.toLowerCase().includes("knockout")) {
+    knockOutFlag = 1;
+    return "Knockout mode activated. Please enter 3 or more players to play";
+  }
+  if (input.toLowerCase().includes("reset")) {
+    playerTurnCounter = 1;
+    totalPlayer = 0;
+    numOfDice = 0;
+    autoModeFlag = 1;
+    reverseModeFlag = 0;
+    knockOutFlag = 0;
+    return "Game reset. Enter number of players to play";
+  }
+
+  if (totalPlayer == 0) {
+    totalPlayer = input;
+    for (let i = 0; i < totalPlayer; i++) {
+      playerDice[i] = [];
+      playerScore[i] = 0;
+    }
+    return `${totalPlayer} players will be playing the game. Please enter the number of dice to roll.`;
+  }
+  if (numOfDice == 0) {
+    numOfDice = input;
+    return `${numOfDice} dice will be rolled for this game. Player 1 please press submit to roll`;
+  }
+  if (knockOutFlag == 1) {
+    return knockOut(input);
+  }
+  return normalGame();
+};
+
+//roll dice
+var diceRoll = function () {
```

Comment:
In Javascript, we usually define the helper functions above and code related to the application logic (i.e. the main function) below. This is because our main function uses all the helper functions, so its convention to define the functions before using them.